### ISSUE TITLE
No gradient defined for operation EluGrad

### ISSUE THREAD

**Issue by altosaar** [tensorflow/tensorflow#4737]
tf.nn.elu does not support second derivatives currently. 

This would be awesome, as in variational autoencoder models these activation functions improve performance by a few nats in the objective compared to tanh, sigmoid (which do support second derivatives).

In the meantime I'm using `elu = lambda x: tf.select(x < 0., tf.exp(x) - 1., x)` which seems to work:

```
In [29]: elu = lambda x: tf.select(x < 0., tf.exp(x) - 1., x)

In [30]: sess.run(tf.gradients(elu(z), z), {z: -1.})
Out[30]: [0.36787945]

In [31]: sess.run(tf.gradients(elu(z), z), {z: 1.})
Out[31]: [1.0]
```

**Comment by tatatodd** [comment]
Contributions are welcome!

**Comment by guotong1988** [comment]
I want to work on it !

**Comment by guotong1988** [comment]
@altosaar What do you mean by 'second derivatives' ? 

I test with the code below and it can not work for version 0.11.

```
x = tf.constant(1)
y = tf.nn.elu(x)
dy = tf.gradients(y,x)

with tf.Session() as sess:
    print sess.run(dy)
```

**Comment by siddharth-agrawal** [comment]
This issue should probably be closed.

**Comment by karmel** [comment]
Automatically closing due to lack of recent activity. Since this issue is old at this point, please reopen the issue if it still occurs when tried with the latest version of Tensorflow. Thank you.
